core/src/main/scala/org/apache/spark/api/python/PythonRDD.scala

```diff
@@ -575,15 +583,32 @@ private[spark] object PythonRDD extends Logging {
     dataOut.write(bytes)
   }
 
-  def writeToFile[T](items: java.util.Iterator[T], filename: String) {
-    import scala.collection.JavaConverters._
-    writeToFile(items.asScala, filename)
-  }
+  private def serveIterator[T](items: Iterator[T]): Int = {
+    val serverSocket = new ServerSocket(0, 1)
+    serverSocket.setReuseAddress(true)
+    serverSocket.setSoTimeout(3000)
+
+    new Thread("serve iterator") {
```

Comment:
Maybe we could add a second `threadName` parameter to this method and pass some more informative identifier to it, such as the RDD id or RDD + partition id.